barretenberg/cpp/src/barretenberg/flavor/flavor.hpp

```diff
@@ -98,6 +98,19 @@ class PrecomputedEntitiesBase {
     uint64_t log_circuit_size;
     uint64_t num_public_inputs;
 };
+// Specifies the regions of the execution trace containing non-trivial wire values
+struct ActiveRegionData {
```

Comment:
don't these ranges need to be non-overlapping and in increasing order? maybe there should be a comment of some sort of check

Reply:
its a good point.  I could add a check on add_range that the input has start >= the previous end. To be safe I suppose I'd also want to make the members private and add getters